topo/node/ceos/ceos_test.go

```diff
@@ -200,106 +250,6 @@ func TestNew(t *testing.T) {
 	}
 }
 
-func TestGenerateSelfSigned(t *testing.T) {
```

Comment:
are there tests in the controller for the generation of self signed certs?

Reply:
I manually validated that it worked and didn't dwell on it (the device does it's own sanity testing when certs are loaded) but I can write some and link the commit back here.

Reply:
Actually, there isn't much I can do to validate the contents of the certificates given the body is effectively random. [This test](https://github.com/aristanetworks/arista-ceoslab-operator/blob/main/test/reconcile-certs/01-assert.yaml) already validates:

- Certificates are configured as secrets
- They are defined as volumes in the pod
- The pod is running

Do to some deficiencies in kuttl I can't validate the volumeMounts are correct: https://github.com/kudobuilder/kuttl/issues/76

The main purpose of this test seems to be to validate KNE's use of the EOS CLI. This isn't relevant from the controller since we're generating them with the go standard `x509` lib. Errors in that are just checked.

The only way a user could mess this up, I suppose, would be to give multiple certs the same name? There isn't much I can really do about that. K8s would report the conflicting mount when creating the pod, and this would show up in the controller's logs. A better approach would be to sanitize the topology (you might already do this, didn't check).

Reply:
Okay I just wanted to make sure we weren't losing coverage here from an overall perspective, sgtm

Reply:
Woops, switch volumes with volumeMounts and vice-versa.